### ISSUE TITLE
Does not handle ternary operator

### ISSUE THREAD

**Issue by sangm** [scalvert/ember-bpr-lint#7]
This should not throw an error

```
environemnt.isBrowser() ? window.bar : 'foo'
```

**Comment by scalvert** [comment]
Looking at this again, it actually should throw an error.

If the ternary condition is true, you should be able to use `window.bar`, otherwise you shouldn't.

Additionally, I'm not even sure if we want to allow ternaries. They will result in some hard to read code. Also, at what point would it be valid to continue without being able to access window APIs?

**Comment by nickiaconis** [comment]
Hey @scalvert, it's been a while; any update on this? I'm bumping into it too. The code with the ternary looks like:

```javascript
return (environment.isBrowser() && window.localStorage) ? window.localStorage : {
  getItem() {
    Ember.Logger.warn(ERROR_ACCESS_STORAGE_OUTSIDE_BROWSER_ENV);
    return '[]';
  },
  setItem() {
    Ember.Logger.warn(ERROR_ACCESS_STORAGE_OUTSIDE_BROWSER_ENV);
  },
  removeItem() {
    Ember.Logger.warn(ERROR_ACCESS_STORAGE_OUTSIDE_BROWSER_ENV);
  },
};
```

Whereas without the ternary it looks like:

```javascript
if (environment.isBrowser()) {
  if (window.localStorage) {
    return window.localStorage;
  }
}

return {
  getItem() {
    Ember.Logger.warn(ERROR_ACCESS_STORAGE_OUTSIDE_BROWSER_ENV);
    return '[]';
  },
  setItem() {
    Ember.Logger.warn(ERROR_ACCESS_STORAGE_OUTSIDE_BROWSER_ENV);
  },
  removeItem() {
    Ember.Logger.warn(ERROR_ACCESS_STORAGE_OUTSIDE_BROWSER_ENV);
  },
};
```

There's also short circuiting going on within the ternary's condition that the rule is not handling properly.

**Comment by mdebbar** [comment]
> Additionally, I'm not even sure if we want to allow ternaries.

@scalvert I agree that ternaries tend to be abused sometimes. But I also believe there are legitimate cases to use ternaries that don't make the code unreadable. And I think the decision of using them or not should be taken by the app owner, not the library author.
```javascript
function safeQuerySelector(selector) {
  return environment.isBrowser() ? document.querySelector(selector) : [];
}
```
```javascript
function isChrome() {
  return environment.isBrowser() ? /chrome/.test(document.body.className) : false;
  // OR:
  const bodyClass = environment.isBrowser() ? document.body.className : '';
  return /chrome/.test(bodyClass);
}
```

**Comment by scalvert** [comment]
You're welcome to submit pull requests!!

**Comment by mdebbar** [comment]
@nickiaconis the lint rule only checks for if statements, it doesn't look at ternary conditions. So I looked into the short-circuiting part of this issue. It seems that the linter works fine given that my test is passing here: #9.

**Comment by nickiaconis** [comment]
@mdebbar I wrote some tests too. I think there's an error in the short circuiting portion (basically, `isBrowser() || window` should be invalid, but passes). #10

**Comment by scalvert** [comment]
Fixed by #14
